app/classifier/tasks/survey/editor.cjsx

```diff
@@ -417,10 +591,83 @@ module.exports = React.createClass
           image.delete().catch =>
             errors += 1
       massDelete.then =>
-        @setState resettingMedia: false
+        @setState resettingFiles: false
         if errors isnt 0
           alert "There were #{errors} errors deleting all those resources. Try again to get the rest."
 
-  handleImportError: (error, file, row) ->
-    @state.importErrors.push {error, file, row}
+  clearInputTypeErrors: (inputType) ->
+    importErrors = @state.importErrors
+    unless importErrors.length is 0
+      newImportErrors = importErrors.filter (error) =>
+        return error if error.inputType isnt inputType
+      @setState importErrors: newImportErrors
+
+  handleImportError: (error, file, inputType, row) ->
+    @state.importErrors.push {error, file, inputType, row}
     @setState importErrors: @state.importErrors
+
+  checkAdd: ->
+    if @state.task.choicesOrder.length is 0
+      true
+    else
+      false
+
+  checkConfusions: ->
+    confusions = false
+    for choiceKey, choiceValue of @state.task.choices
+      if choiceValue.confusionsOrder.length isnt 0
+        confusions = true
+    confusions
+
+  checkClear: ->
+    if (@state.choicesName or @state.characteristicsName or @state.confusionsName or @state.questionsName)
+      false
+    else
+      true
+
+  checkApply: ->
+    if not @checkClear() and @state.importErrors.length is 0
+      false
+    else
+      true
+
+  checkContentWarnings: ->
```

Comment:
this method is very long and should be refactored for maintainability (not needed for merge)